### PR TITLE
Add MSA backward compatibility tests

### PR DESCRIPTION
**Pull request by philandstuff**
This adds tests for MSA backward compatibility.  They reproduce the
old ida-hub-acceptance-tests/msa-backward-compatibility-test.sh which
ran
MsaBackwardsCompatibilityTest.userSignInSuccessfullyAndCompletesNINOCycle3Page.

Basically it's a simple login flow with cycle 3.

The purpose of these tests is that they are run against a separate
test-rp/msa instance, which are running an old but supported version
of the MSA.  These MSAs are already deployed to PaaS and have
pipelines to deploy them.

It's not ideal to hardcode the MSA URL in the feature but I couldn't
think of a better way to achieve this.  I'm open to suggestions.

I have run these locally and they pass :+1: so we are still backwards
compatible, hooray!

### CODE REVIEW

**Comment by galund** [comment]
Can you link us the PRs that set up the three MSA instances? ta

**Comment by philandstuff** [comment]
@galund 
https://github.com/alphagov/verify-hub-federation-config/pull/496
https://github.com/alphagov/verify-matching-service-adapter/pull/155
https://github.com/alphagov/verify-test-rp/pull/43
https://github.com/alphagov/verify-infrastructure-config/pull/264
https://github.com/alphagov/verify-infrastructure-config/pull/267
😅

File: features/backward_compatibility.feature [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    And they select IDP "Stub Idp Demo Two"
+    And they login as "stub-idp-demo-two-c3"
+    And they submit cycle 3 "AA123456A"
+    Then they should be successfully verified
+
+  Examples:
+    | RP                                                              |
+    | https://test-rp-staging-backcompat-1.cloudapps.digital/test-rp/ |
+    | https://test-rp-staging-backcompat-2.cloudapps.digital/test-rp/ |
+    | https://test-rp-staging-backcompat-3.cloudapps.digital/test-rp/ |
```
Comment (CharlesIC) [inline features/backward_compatibility.feature]: We're going to have to remember to add new ones here, right? I guess if 4 and 5 are currently unoccupied that's _fine_

Comment (philandstuff) [reply features/backward_compatibility.feature]: Yeah I don't know cucumber well enough to find a way to avoid this toil :(

**Pull request merged**